addons/website/static/src/snippets/s_social_media/options.js

```diff
@@ -0,0 +1,127 @@
+/** @odoo-module **/
+
+import options from 'web_editor.snippets.options';
+import {SocialMedia} from '@web_editor/js/editor/snippets/s_social_media/options'
+
+let dbSocialValues;
+let dbSocialValuesProm;
+options.registry.SocialMediaWebsite = SocialMedia.extend({
+    /**
+     * @override
+     */
+    async cleanForSave() {
+        // Update the DB links.
+        let websiteId;
+        this.trigger_up('context_get', {
+            callback: function (ctx) {
+                websiteId = ctx['website_id'];
+            },
+        });
+        const fields = {};
+        for (const [mediaName, url] of Object.entries(dbSocialValues)) {
+            fields["social_" + mediaName] = url;
+        }
+        await this._rpc({
+            model: 'website',
+            method: 'write',
+            args: [[websiteId], fields],
+        });
+    },
+
+
+    //--------------------------------------------------------------------------
+    // Private
+    //-------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    async _computeWidgetState(methodName, params) {
+        const _super = this._super.bind(this);
+        if (methodName === 'renderListItems') {
+            await this._fetchSocialMedia();
+        }
+        const result = _super(methodName, params);
+        for (const entry of this._lastEntries) {
+            if (entry.media) {
+                dbSocialValues[entry.media] = entry.display_name;
+            }
+        }
+        return result;
+    },
```

Comment:
I don't understand what `_lastEntries` means nor its purpose. The _computeWidgetState method is supposed to infer information from the DOM and/or the database. If you keep track of the values in JS, it's not ok. In the original implementation, we had `entriesNotInDom` already breaking that rule but for exceptional reasons... is it the same here? If yes, why renamed "last entries" ?

Also, updating dbSocialValues in `_computeWidgetState` cannot be right, it's not supposed to update anything, just saying what the current values are, checking the DOM and/or database.